src/core/export.js

```diff
@@ -0,0 +1,303 @@
+var moment = require('moment/moment');
+var _ = require('underscore');
+import XLSX from 'xlsx';
+import * as filesaver from 'filesaver.js';
+import axios from 'axios';
+import urljoin from 'url-join';
+
+/*
+ * Takes current data displayed in the DataTable, Annual Cycle graph, or
+ * Projected Change graph, along with contextual data from user input, creates
+ * an XLSX or CSV file, and serves it to the user for download. Draws on example
+ * code from js-xlsx docs: https://github.com/SheetJS/js-xlsx
+ */
+var exportDataToWorksheet = function(datatype, metadata, data, format, timeidx) {
+  // create workbook object containing one or more worksheets
+  var wb = {
+      Sheets: {},
+      SheetNames: []
+  };
+
+  // prepare filename, metadata cells, and data cells according to type of export
+  var summaryCells, dataCells, outputFilename;
+  var filenamePrefix = "PCIC_CE_";
+  var filenameInfix = "Export_" + metadata.model_id + "_" + metadata.experiment + "_" + metadata.variable_id;
+  var filenameSuffix = "." + format;
+  switch(datatype) {
+    case "timeseries":
+      summaryCells = createTimeSeriesWorksheetSummaryCells(metadata);
+      dataCells = fillMultiTimeSeriesWorksheetDataCells(data, metadata);
+      outputFilename = `${filenamePrefix}TimeSeries${filenameInfix}${filenameSuffix}`;
+      break;
+    case "stats":
+      summaryCells = createWorksheetSummaryCells(metadata, timeIndexToTimeOfYear(timeidx));
+      dataCells = fillWorksheetDataCells(data);
+      outputFilename = `${filenamePrefix}StatsTable${filenameInfix}_${timeIndexToTimeOfYear(timeidx)}${filenameSuffix}`;
+      break;
+    case "climoseries":
+      summaryCells = createWorksheetSummaryCells(metadata, timeIndexToTimeOfYear(timeidx));
+      dataCells = fillClimoSeriesDataCells(data);
+      outputFilename = `${filenamePrefix}ProjectedChange${filenameInfix}_${timeIndexToTimeOfYear(timeidx)}${filenameSuffix}`;
+      break;
+    case "single-timeseries":
+      summaryCells = createTimeSeriesWorksheetSummaryCells(metadata);
+      dataCells = fillSingleTimeSeriesWorksheetDataCells(data, metadata);
+      outputFilename = `${filenamePrefix}TimeSeries${filenameInfix}${filenameSuffix}`;
+      break;
+  }
+
+  // assemble the worksheet and add it to the workbook.
+  // Note: sheetName will be truncated to 31 chars in
+  // XLSX export to meet Excel limitation
+  var ws = assembleWorksheet(summaryCells.concat([[]], dataCells));
+  var sheetName = `${datatype}_${metadata.variable_id}`;
+  wb.SheetNames.push(sheetName);
+  wb.Sheets[sheetName] = ws;
+
+  function xml_to_binary_string(s) {
```

Comment:
What's going on here? Or more specifically, why do we need to do this? (Not that I doubt that we need to, but a short comment of the "why" would be sufficiently helpful).